wgsl/wgsl_unit_tests.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+#
+# Copyright 2023 Google LLC
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+# 1. Redistributions of works must retain the original copyright
+# notice, this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the original
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#
+# 3. Neither the name of the W3C nor the names of its contributors
+# may be used to endorse or promote products derived from this work
+# without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+import argparse
+import os
+import sys
+from tree_sitter import Language, Parser
+
+SCRIPT='wgsl_unit_tests.py'
+
+class Case:
+    """
+    A test case
+    """
+    def __init__(self,text,expect_pass=True):
+        self.text = text
+        self.expect_pass = (expect_pass == True)
+
+    def __str__(self):
+        expectation = "expect_pass" if self.expect_pass else "expect_fail"
+        return "Case:{}:\n---\n{}\n---".format(expectation,self.text)
+
+class XFail(Case):
+    def __init__(self,text):
+        super().__init__(text,expect_pass=False)
+
+cases = [
```

Comment:
Suggest adding a few more cases to cover comments (both styles) in/out of templates (may or may not contain some template-looking characters.

Reply:
Definitely.  The custom scanner has some TODOs to resolve.  Will do so after landing.